src/tetris/Tetromino.java

```diff
@@ -0,0 +1,354 @@
+            for (int i = 0; i < squares.size(); i++) {
+                squares.get(i).setX(rotationPointX + rotationPointY - oldY.get(i));
+                squares.get(i).setY(rotationPointY - rotationPointX + oldX.get(i));
+            }
+            drawShape();
+        }
+    }
+
+     /**
+     * This method returns true if the new position of each square of the tetrimono after
+     * rotation is out of bounds.
+     */
+    private Boolean checkWallCollisionAfterRotation(Board board){
+        List<Integer> oldY = getOldYs();
+        List<Integer> oldX = getOldXs();
+        for (int i = 0; i < squares.size(); i++) {
+            if  (rotationPointX + rotationPointY - oldY.get(i) < 0
+                || rotationPointX + rotationPointY - oldY.get(i) >= board.getWidth()
+                || rotationPointY - rotationPointX + oldX.get(i) >= board.getGrid().size()
+                || rotationPointY - rotationPointX + oldX.get(i) < 0){
+                    return true;
+                }
+        }
+        return false;
+    }
+
+    /**
+     * This method returns true if the new position of each square of the current tetrimono after
+     * rotation is occupied by other tetrominoes already added to the board.
+     */
+    private Boolean checkBoardCollisionAfterRotation(Board board){
+        List<Integer> oldX = getOldXs();
+        List<Integer> oldY = getOldYs();
+        for (int i = 0; i < squares.size(); i++) {
+            if (board.getGrid().get(rotationPointY - rotationPointX + oldX.get(i)).get(rotationPointX + rotationPointY - oldY.get(i))){
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * This methods returns an unmodifiable list of the current x position
+     * of each square that makes up the tetromino.
+     * 
+     */
+    private List<Integer> getOldXs() {
+        List<Integer> result = new ArrayList<>();
+        for (Square square : squares) {
+            result.add(square.getX());
+        }
+        return Collections.unmodifiableList(result);
+    }
+    /**
+     * This methods returns an unmodifiable list of the current y position
+     * of each square that makes up the tetromino.
+     * 
+     */
+    private List<Integer> getOldYs() {
+        List<Integer> result = new ArrayList<>();
+        for (Square square : squares) {
+            result.add(square.getY());
+        }
+        return Collections.unmodifiableList(result);
+    }
+
+    /**
+     * This methods add 4 square objects to the list of squares to make
+     * a tetromino of type z Shape 1.
+     * 
+     */
+    private void zShape1() {
+        squares.add(new Square(0, 0, new Rectangle(0, 0, squareSize, squareSize)));
+        squares.add(new Square(1, 0, new Rectangle(0, 0, squareSize, squareSize)));
+        squares.add(new Square(1, 1, new Rectangle(0, 0, squareSize, squareSize)));
+        squares.add(new Square(2, 1, new Rectangle(0, 0, squareSize, squareSize)));
```

Comment:
The repetition of `new Rectangle(0, 0, squareSize, squareSize)` throughout suggests that you may not have quite the right abstraction yet. (Different Square constructor maybe?)

If I were preparing this code for future growth, I’d definitely move all these shapes out of (repetitive) code and into data somehow.